src/circuit.rs

```diff
@@ -187,12 +237,13 @@ pub trait Layouter<C: Chip> {
     ///
     /// ```ignore
     /// fn assign_region(&mut self, || "region name", |region| {
-    ///     region.assign_advice(self.config.a, offset, || { Some(value)});
+    ///     let config = chip.config().clone();
```

Comment:
I don't believe the `.clone()` should be necessary, now that `chip` and `layouter` are separate variables (so we aren't trying to use an immutable reference in a method that takes a mutable reference to `self`).